### PR TITLE
Category, group and subgroup support

### PR DESCRIPTION
**Pull request by NullMember**
Basic category, group and subgroup support. Group and subgroup automatically hides nodes under them.

### CODE REVIEW

**Comment by 4d49** [comment]
I found the problem. If you collapse/expand a group, subgroups will be expanded, but the button state will not be pressed and will need to be pressed twice.

**Comment by NullMember** [comment]
Last commit should fix the issue

**Comment by 4d49** [comment]
The `test_1` property is always hidden in the inspector .

```gdscript
@export_category("Сategory")

@export_group("The property is always hidden")
@export var test_1: bool = true

@export_subgroup("In a subgroup, everything works")
@export var test_2: bool = true
```

**Comment by 4d49** [comment]
A radical solution to the problem would be to add properties as child nodes to categories/groups. But then there is the problem of finding properties.

File: scripts/inspector.gd [hunk 1]
```diff
@@ -137,11 +144,21 @@ func update_inspector(filter: String = _search.text) -> void:
...
-		if filter.is_subsequence_ofn(property["name"]) and is_valid_property(property):
-			var property_control = create_property_control(_object, property)
-			if is_instance_valid(property_control):
-				_container.add_child(property_control)
+		if _start_populate:
+			if filter.is_subsequence_ofn(property["name"]) and is_valid_property(property):
+				var property_control = create_property_control(_object, property)
+				if is_instance_valid(property_control):
+					_container.add_child(property_control)
+		if property["name"].ends_with(".gd"):
```
Comment (4d49) [inline scripts/inspector.gd:155]:
```suggestion
		elif property["name"].ends_with(".gd"):
```
To exclude the check if we've already started.

**Review by 4d49 — CHANGES_REQUESTED** [review]

**Pull request merged**